fh-messaging/lib/mongo_functions/metrics_r.js

```diff
@@ -2,30 +2,70 @@
 //ts and emit are used for testing purposes
 var ts;
 var emit;
-function metrics_r(k, vals) {
-  // Take the domain and ts in the first vals entry as the reduced elements domain and ts
-  var ret = {}, i, p, tempVal;
-
-  for (i in vals) {
-    if (vals.hasOwnProperty(i)) {
-      tempVal = vals[i];
-      for (p in tempVal) {
-        if ('undefined' !== typeof ret[p]) {
-          ret[p] += tempVal[p];
-        } else {
-          ret[p] = tempVal[p];
+
+function metrics_r(k, metrics) {
+
+  function groupById(values) {
+    var groups = {};
+    for (index in values) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add the "var" keyword to this declaration of "index". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AS2703)